mlir/python/mlir/dialects/scf.py

```diff
@@ -132,8 +132,8 @@ def for_(
     iter_args = tuple(for_op.inner_iter_args)
     with InsertionPoint(for_op.body):
         if len(iter_args) > 1:
-            yield iv, iter_args
+            yield iv, iter_args, for_op.results
         elif len(iter_args) == 1:
-            yield iv, iter_args[0]
+            yield iv, iter_args[0], for_op.results
```

Comment:
I've actually never double checked this but I believe for a single iter args, the result my also be "single"? Thus, you can do this
```suggestion
            yield iv, iter_args[0], for_op.results[0]
```

Reply:
I thought I was returning `results[0]`, thanks for the catch. But the test actually worked without that :)